### PR TITLE
Add 0.10.0 change log

### PR DESCRIPTION
**Pull request by RobertIndie**
### Motivation

Add 0.10.0 change log

**Note: This PR shouldn't be merged until the release vote is passed.**

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / GoDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Review by mattisonchao — APPROVED** [review]

**Review by shibd — APPROVED** [review]

**Review by maxsxu — APPROVED** [review]
Great to see 0.10.0 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,92 @@
 
 All notable changes to this project will be documented in this file.
 
+[0.10.0] 2023-03-27
+
+## Feature
+* Support chunking for big messages. by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/805
+* Add BackoffPolicy to `reader` and improve test case by @labuladong in https://github.com/apache/pulsar-client-go/pull/889
+* Support cumulative acknowledge by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/903
```
Comment (momo-jun) [inline CHANGELOG.md:31]:
```suggestion
* Support cumulative acknowledgment by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/903
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,92 @@
...
+* Support batch index ACK by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/938
+* Support Exclusive Producer access mode. by @shibd in https://github.com/apache/pulsar-client-go/pull/944
+* Add transactionCoordinatorClient by @liangyepianzhou in https://github.com/apache/pulsar-client-go/pull/953
+* Support memory limit for producer. by @shibd in https://github.com/apache/pulsar-client-go/pull/955
+* Support grouping ACK requests by time and size by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/957
+* Support WaitForExclusive producer access mode. by @shibd in https://github.com/apache/pulsar-client-go/pull/958
+* Support Copper Argos in the Athenz auth provider by @massakam in https://github.com/apache/pulsar-client-go/pull/960
+* Support auto release idle connections by @RobertIndie in https://github.com/apache/pulsar-client-go/pull/963
+* Support batch index ACK and max number of messages in batch by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/967
+* Support auto scaled consumer receiver queue by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/976
```
Comment (momo-jun) [inline CHANGELOG.md:43]:
```suggestion
* Support auto-scaled consumer receiver queue by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/976
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,6 +23,92 @@
...
+* Implement transactionImpl by @liangyepianzhou in https://github.com/apache/pulsar-client-go/pull/984
+* Expose the chunk config of consumer to the reader by @CrazyCollin in https://github.com/apache/pulsar-client-go/pull/987
+* Support consumer client memory limit by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/991
+
+
+## Improve
+* Nack the message in dlqrouter when send errors by @leizhiyuan in https://github.com/apache/pulsar-client-go/pull/592
```
Comment (momo-jun) [inline CHANGELOG.md:50]:
```suggestion
* Nack the message in dlqrouter when sending errors by @leizhiyuan in https://github.com/apache/pulsar-client-go/pull/592
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -23,6 +23,92 @@
...
+* Fix typo in consumer.go by @sekfung in https://github.com/apache/pulsar-client-go/pull/857
+* Fix marshalling time.Time{} to uint64 by @aymkhalil in https://github.com/apache/pulsar-client-go/pull/865
+* Use the `DATA` constant as prefix in OAuth2 KeyFileProvider by @Niennienzz in https://github.com/apache/pulsar-client-go/pull/866
+* Fix bot cannot get the pr link by @RobertIndie in https://github.com/apache/pulsar-client-go/pull/868
+* Fix PR template by @RobertIndie in https://github.com/apache/pulsar-client-go/pull/869
+* Add go test flag '-v' for more clearly CI log by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/871
+* Fix the dispatcher() stuck caused by availablePermitsCh by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/875
+* Fix the Send() stuck caused by callback() not being called by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/880
+* Fix the data race of ackReq.err by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/881
+* Add data url format to read the key file by @nodece in https://github.com/apache/pulsar-client-go/pull/883
```
Comment (momo-jun) [inline CHANGELOG.md:63]:
```suggestion
* Add data URL format to read the key file by @nodece in https://github.com/apache/pulsar-client-go/pull/883
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -23,6 +23,92 @@
...
+* Prevent consumer panic on de-serializing message if schema not found by @GPrabhudas in https://github.com/apache/pulsar-client-go/pull/886
+* Fix load tls certificate by @nodece in https://github.com/apache/pulsar-client-go/pull/888
```
Comment (momo-jun) [inline CHANGELOG.md:65]:
```suggestion
* Fix the conditions of loading TLS certificate by @nodece in https://github.com/apache/pulsar-client-go/pull/888
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -23,6 +23,92 @@
...
+* Handle nil value message correctly in table-view by @Demogorgon314 in https://github.com/apache/pulsar-client-go/pull/930
+* Migrate from the deprecated io/ioutil package by @reugn in https://github.com/apache/pulsar-client-go/pull/942
+* Update the Cobra library to significantly reduce the dependency tree by @reugn in https://github.com/apache/pulsar-client-go/pull/943
+* Remove go1.11 code leftovers by @reugn in https://github.com/apache/pulsar-client-go/pull/946
+* Use pkg.go.dev badge in the readme by @reugn in https://github.com/apache/pulsar-client-go/pull/947
+* Improve test script by @nodece in https://github.com/apache/pulsar-client-go/pull/951
+* Optimize the performance by passing MessageID implementations by pointers by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/968
+* Fix flaky Key_Shared subscription related tests by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/970
+* Refactor the toTrackingMessageID() by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/972
+* Prevent rpc client panic on rpc response if `ProducerReady` is nil by @sekfung in https://github.com/apache/pulsar-client-go/pull/973
```
Comment (momo-jun) [inline CHANGELOG.md:84]:
```suggestion
* Prevent RPC client panic on RPC response if `ProducerReady` is nil by @sekfung in https://github.com/apache/pulsar-client-go/pull/973
```
BTW, is 'nil' a correct value or short for something else?

File: CHANGELOG.md [hunk 7]
```diff
@@ -23,6 +23,92 @@
...
+* Remove the outdated interface description of `SeekByTime` by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/924
+* Handle nil value message correctly in table-view by @Demogorgon314 in https://github.com/apache/pulsar-client-go/pull/930
+* Migrate from the deprecated io/ioutil package by @reugn in https://github.com/apache/pulsar-client-go/pull/942
+* Update the Cobra library to significantly reduce the dependency tree by @reugn in https://github.com/apache/pulsar-client-go/pull/943
+* Remove go1.11 code leftovers by @reugn in https://github.com/apache/pulsar-client-go/pull/946
+* Use pkg.go.dev badge in the readme by @reugn in https://github.com/apache/pulsar-client-go/pull/947
+* Improve test script by @nodece in https://github.com/apache/pulsar-client-go/pull/951
+* Optimize the performance by passing MessageID implementations by pointers by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/968
+* Fix flaky Key_Shared subscription related tests by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/970
```
Comment (momo-jun) [inline CHANGELOG.md:82]:
```suggestion
* Fix flaky Key_Shared subscription-related tests by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/970
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -23,6 +23,92 @@
...
+* Support batch index ACK and max number of messages in batch by @BewareMyPower in https://github.com/apache/pulsar-client-go/pull/967
+* Support auto scaled consumer receiver queue by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/976
+* Implement transactionImpl by @liangyepianzhou in https://github.com/apache/pulsar-client-go/pull/984
+* Expose the chunk config of consumer to the reader by @CrazyCollin in https://github.com/apache/pulsar-client-go/pull/987
+* Support consumer client memory limit by @Gleiphir2769 in https://github.com/apache/pulsar-client-go/pull/991
+
+
+## Improve
+* Nack the message in dlqrouter when send errors by @leizhiyuan in https://github.com/apache/pulsar-client-go/pull/592
+* Fix tls for certificate that does not include IP SANS, save hostname before switching to physical addr by @dinghram in https://github.com/apache/pulsar-client-go/pull/812
```
Comment (momo-jun) [inline CHANGELOG.md:51]:
```suggestion
* Fix TLS certificates that do not include IP SANS, save hostname before switching to a physical address by @dinghram in https://github.com/apache/pulsar-client-go/pull/812
```

File: CHANGELOG.md (hunk 6 above)
Comment (RobertIndie) [reply CHANGELOG.md:84]: Yes, it represents `null` in go.

**Review by momo-jun — APPROVED** [review]
LGTM.

**Pull request merged**